src/test/java/teammates/test/cases/datatransfer/AccountAttributesTest.java

```diff
@@ -139,6 +141,65 @@ public void testLegacyAccountEntityToAttributes() {
 
     }
 
+    @Test
+    public void testBuilderWithDefaultValues() {
+        AccountAttributes observedAccountAttributes = AccountAttributes.builder().build();
+
+        assertEquals(null, observedAccountAttributes.createdAt);
+        assertEquals(null, observedAccountAttributes.getEmail());
+        assertEquals(null, observedAccountAttributes.getGoogleId());
+        assertEquals(null, observedAccountAttributes.getInstitute());
+        assertFalse(observedAccountAttributes.isInstructor());
+        assertEquals(null, observedAccountAttributes.getName());
+        assertEquals(null, observedAccountAttributes.studentProfile);
+    }
+
+    @Test
+    public void testBuilderWithPopulatedFieldValues() {
+        String expectedGoogleId = "dummyGoogleId";
+        String expectedEmail = "dummyEmail";
+        String expectedName = "dummyName";
+        String expectedInstitute = "dummyInstitute";
+        boolean expectedIsInstructor = true; //since false case is covered in default test
+        Date expectedCreatedAt = new Date(98765);
+
+        AccountAttributes observedAccountAttributes = AccountAttributes.builder()
+                .withGoogleId(expectedGoogleId)
+                .withEmail(expectedEmail)
+                .withName(expectedName)
+                .withInstitute(expectedInstitute)
+                .withIsInstructor(expectedIsInstructor)
+                .withCreatedAt(expectedCreatedAt)
+                .withDefaultStudentProfileAttributes(expectedGoogleId)
+                .build();
+
+        assertEquals(expectedGoogleId, observedAccountAttributes.getGoogleId());
+        assertEquals(expectedEmail, observedAccountAttributes.getEmail());
+        assertEquals(expectedCreatedAt, observedAccountAttributes.createdAt);
+        assertEquals(expectedInstitute, observedAccountAttributes.getInstitute());
+        assertEquals(expectedIsInstructor, observedAccountAttributes.isInstructor());
+        assertEquals(expectedName, observedAccountAttributes.getName());
+        assertEquals(expectedGoogleId, observedAccountAttributes.studentProfile.googleId);
+    }
+
+    @Test
+    public void testBuilderWithUnsanitisedFieldValues() {
+        AccountAttributes observedAccountAttributes = AccountAttributes.builder()
+                .withGoogleId("googleId@gmail.com")
+                .withName("  random  name with   extra spaces    ")
+                .withEmail("         email   ")
```

Comment:
How about trying something like `"&<email>&"` rather than just spaces for all types of sanitization?

Reply:
I am not sure what "&\<email\>&" is.
tried looking it up but could not find much.
Can u please point to some resources?

Reply:
Sorry my bad. I assumed the sanitization was for HTML when it actually wasn't.
You can ignore this thread

Reply:
Final change here and then it looks perfect to me ^

Reply:
@lightt77 We want to provide valid email address here.